File: java/src/main/java/com/tyouqu/database/FullTableOperationInterceptor.java
package com.tyouqu.database;

public final class FullTableOperationInterceptor implements SqlInterceptor {
    private final SqlSafetyOptions options;

    public FullTableOperationInterceptor(SqlSafetyOptions options) {
        this.options = options == null ? SqlSafetyOptions.defaults() : options;
    }

    @Override
    public void beforeExecute(SqlExecutionContext context) {
        String normalized = SqlSafetyAnalyzer.normalize(context.sql());
        if (options.blockFullTableUpdate() && SqlSafetyAnalyzer.isFullTableUpdate(normalized)) {
            throw new TyouquDatabaseException("Unsafe SQL was blocked: UPDATE without WHERE is not allowed.")
                .provider(context.provider().name())
                .sqlId(context.sqlId());
        }
        if (options.blockFullTableDelete() && SqlSafetyAnalyzer.isFullTableDelete(normalized)) {
            throw new TyouquDatabaseException("Unsafe SQL was blocked: DELETE without WHERE is not allowed.")
                .provider(context.provider().name())
                .sqlId(context.sqlId());
        }
    }
}

File: dotnet/src/Tyouqu.Database.Abstractions/FullTableOperationInterceptor.cs
namespace Tyouqu.Database.Abstractions;

public sealed class FullTableOperationInterceptor : ISqlInterceptor
{
    private readonly SqlSafetyOptions _options;

    public FullTableOperationInterceptor(SqlSafetyOptions options)
    {
        _options = options;
    }

    public void BeforeExecute(SqlExecutionContext context)
    {
        var normalized = SqlSafetyAnalyzer.Normalize(context.Sql);
        if (_options.BlockFullTableUpdate && SqlSafetyAnalyzer.IsFullTableUpdate(normalized))
        {
            throw new TyouquDatabaseException("Unsafe SQL was blocked: UPDATE without WHERE is not allowed.")
            {
                Provider = context.Provider.ToString(),
                SqlId = context.SqlId
            };
        }

        if (_options.BlockFullTableDelete && SqlSafetyAnalyzer.IsFullTableDelete(normalized))
        {
            throw new TyouquDatabaseException("Unsafe SQL was blocked: DELETE without WHERE is not allowed.")
            {
                Provider = context.Provider.ToString(),
                SqlId = context.SqlId
            };
        }
    }
}
